src/containers/BookList.js

```diff
@@ -2,41 +2,50 @@ import React from 'react';
 import PropTypes from 'prop-types';
 import { connect } from 'react-redux';
 import Book from '../components/Book';
+import CategoryFilter from '../components/CategoryFilter';
+import { removeBook } from '../actions/index';
 
 const mapDispatchToProps = dispatch => ({
   removeBook: book => {
-    dispatch({
-      ID: book.ID,
-      type: 'REMOVE_BOOK',
-    });
+    dispatch(removeBook(book.ID));
   },
 });
 
-const mapStateToProp = state => ({ books: state.bookReducer });
+const mapStateToProp = state => ({
+  books: state.bookReducer,
+  filter: state.filterReducer,
+});
 
-const BookList = ({ books, removeBook }) => {
+const BookList = ({ books, removeBook, filter }) => {
   const handleRemoveBook = ID => {
     removeBook(ID);
   };
 
   const renderBook = book => <Book key={book.ID} book={book} removeBook={handleRemoveBook} />;
+
+  const filteredBooks = filter === 'All' ? books : books.filter(book => book.category === filter);
+
   return (
-    <table>
-      <thead>
-        <tr>
-          <th>ID</th>
-          <th>title </th>
-          <th>category</th>
-        </tr>
-      </thead>
-      <tbody>{books.map(renderBook)}</tbody>
-    </table>
+    <div>
+      <CategoryFilter />
```

Comment:
Kindly pass the handleFilterChange down to the categoryFilter

Reply:
Done.